File: Difficulty/Easy/[119] Pascal's Triangle II/solution.cpp
/*
 * @lc app=leetcode id=119 lang=cpp
 *
 * [119] Pascal's Triangle II
 */

// @lc code=start
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        if (rowIndex == 0) return vector<int>{1};
        if (rowIndex == 1) return vector<int>{1, 1};
        if (rowIndex == 2) return vector<int>{1, 2, 1};
        vector<int> res(rowIndex+1, 0);
        res[0] = 1;
        res[1] = 2;
        res[2] = 1;
        for (int i = 3; i <= rowIndex; i++){
            int prev = 1, tmp;
            for (int j = 1; j < i; j++){
                tmp = res[j] + prev;
                prev = res[j];
                res[j] = tmp;
            }
            res[i] = 1;
        }
        return res;
    }
};
// @lc code=end

File: Difficulty/Easy/[119] Pascal's Triangle II/solution.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if rowIndex == 0: return [1]
        if rowIndex == 1: return [1, 1]
        if rowIndex == 2: return [1, 2, 1]
        res = [0] * (rowIndex + 1)
        res[0], res[1], res[2] = 1, 2, 1
        for i in range(3, rowIndex + 1):
            prev, tmp = 1, 0
            for j in range(1, i):
                tmp = res[j] + prev
                prev = res[j]
                res[j] = tmp
            res[i] = 1
        return res